website/blog/2018-07-14-whats-the-deal-with-the-new-pipeline-plugin-configuration.md

```diff
@@ -0,0 +1,153 @@
+
+As currently specced, arrow functions are required to be wrapped in parentheses:
+
+```js
+let person = { score: 25 };
+
+let newScore = person.score
+  |> double
+  |> (_ => add(7, _))
+  |> (_ => boundScore(0, 100, _));
+```
+
+Exploration is underway to determine whether it would be feasible to enable arrow functions to be used without parentheses, as they are a signficant syntactical burden.
+
+On the question of async, F# Pipelines treat `await` similar to a unary function:
+
+```js
+promise |> await
+```
+
+This would desugar to:
+
+```js
+await promise
+```
+
+and can thus be used in the middle of larger function chains with async:
+
+```js
+promise
+  |> await
+  |> (x => doubleSay(x, ', '))
+  |> capitalize
+  |> (x => x + '!')
+  |> (x => new User.Message(x))
+  |> (x => stream.write(x))
+  |> await
+  |> console.log;
+```
+
+The special casing of `await` could potentially enable other unary operators to be used similarly (e.g. `typeof`), but the F# pipelines don't support them initially.
+
+### [Smart Pipelines](https://github.com/js-choi/proposal-smart-pipelines/)
+
+Smart Pipelines takes the idea of the placeholder to its logical conclusion, enabling it to manage partial application as well as arbitary expressions in a pipeline. The above long chain would be written thus:
+
+```js
+promise
+  |> await #
+  |> doubleSay(#, ', ')
+  |> # || throw new TypeError()
+  |> capitalize
+  |> # + '!'
+  |> new User.Message(#)
+  |> await stream.write(#)
+  |> console.log;
+```
+
+In ["topic style"](https://github.com/js-choi/proposal-smart-pipelines/blob/master/readme.md#topic-style), a placeholder (called a "lexical topic token") is required, and the code will throw an early SyntaxError if it is not included.
+
+```js
+10 |> # + 1;
+promise |> await #;
+```
+
+In ["bare style"](https://github.com/js-choi/proposal-smart-pipelines/blob/master/readme.md#bare-style), where a bare identifier is provided to a step in the pipeline, no token is necessary.
+
+```js
+x |> a;
+x |> f.b;
+```
+
+If there are any operators, parentheses (including for method calls), brackets, or anything other than identifiers and dot punctuators, then it is in topic style, not in bare style, and a topic token is necessary. This is avoids footguns and eliminate ambiguity in bare style code.
+
+Smart pipelines thus resolve the issue of async in an integrative way, allowing all possible expressions to be emedded in a pipeline; not only `await`, but also `typeof`, `yield`, and another other operator desired.
+
+## Where Babel Comes In
+
+Once all three proposals were fleshed out, we realized discussion and debate were unlikely to resolve the inherent tension between them. We decided the best way to make a decision would be with the feedback of developers, using the proposals in real code. Given babel's role in the community, we approached them about including all three of the above proposals into the pipeline operator plugin.
```

Comment:
Who is we? I think there are multiple we's in the post that could be made clear

Reply:
I suppose "we" here could be @js-choi and I, specifically, since we're the one's doing the implementation, but it's kind of intentionally ambiguous because there are a lot of people involved in this, and some of them are involved in both TC39 & Babel, so it's not really clear who the "we" actually _is_ here as a group / label.

Reply:
I just thought it might get confusing with `we approached them` and stuff

Reply:
You're right, maybe I rephrase it as: "Given Babel's role in the community, we decided to include all three proposals..." so it reads more like it's the same group? It's still a bit weird, but maybe better?

Reply:
Ok, made this change.